Bottom: 09fa862d8c12899825303243076d24d3b121a4e5
Top:    5955592b21d684213b510d83627ae9ac35cd5ec7
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-09-14 21:40:27 +0100

gtk: Fix ghid_zoom_view_fit() to work when the board is flipped.

Affects-bug: lp-850311


---

diff --git a/src/hid/gtk/gtkhid-main.c b/src/hid/gtk/gtkhid-main.c
index 4383ee5..e0cfbc9 100644
--- a/src/hid/gtk/gtkhid-main.c
+++ b/src/hid/gtk/gtkhid-main.c
@@ -128,9 +128,10 @@ ghid_zoom_view_rel (Coord center_x, Coord center_y, double factor)
 static void
 ghid_zoom_view_fit (void)
 {
-  ghid_pan_view_abs (0, 0, 0, 0);
-  ghid_zoom_view_abs (0, 0, MAX (PCB->MaxWidth  / gport->width,
-                                 PCB->MaxHeight / gport->height));
+  ghid_pan_view_abs (SIDE_X (0), SIDE_Y (0), 0, 0);
+  ghid_zoom_view_abs (SIDE_X (0), SIDE_Y (0),
+                      MAX (PCB->MaxWidth  / gport->width,
+                           PCB->MaxHeight / gport->height));
 }
 
 static void
